sdk/compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Upgrade.hs

```diff
@@ -88,9 +88,17 @@ extractDiagnostics version upgradeInfo action =
     Left err -> [toDiagnostic err]
     Right ((), warnings) -> map toDiagnostic (nub warnings)
 
+unitIdDalfPackageToUpgradedPkg :: (UnitId, LF.DalfPackage) -> UpgradedPkgWithNameAndVersion
+unitIdDalfPackageToUpgradedPkg (unitId, dalfPkg) = 
+  let (packageName, mbPkgVersion) = LF.splitUnitId unitId
+      LF.DalfPackage{dalfPackagePkg,dalfPackageId=presentPkgId} = dalfPkg
+      presentPkg = extPackagePkg dalfPackagePkg
+  in
+  (presentPkgId, presentPkg, packageName, mbPkgVersion)
+
 checkPackage
   :: LF.Package
-  -> [(UnitId, LF.DalfPackage)] -> Version -> UpgradeInfo
+  -> [UpgradedPkgWithNameAndVersion] -> Version -> UpgradeInfo
```

Comment:
Since it  is in the API of the module, why not making `UpgradedPkgWithNameAndVersion` a proper type instead of an alias.

Reply:
Fixed